addons/website/static/src/snippets/s_share/000.js

```diff
@@ -77,7 +76,13 @@ const ShareWidget = publicWidget.Widget.extend({
                 return a + title + c;
             })
             .replace(this.MEDIA_REGEX, (match, a, b, c) => {
-                return a + media + c;
+                // Some pages (/profile/user/ID) don't have an image to share.
+                const ogImageEl = document.querySelector("meta[property='og:image']");
+                if (ogImageEl) {
+                    const media = encodeURIComponent(ogImageEl.content);
+                    return a + media + c;
+                }
+                return a.substring(0, a.lastIndexOf("&media=")) + c;
```

Comment:
If `media` is the first URL parameter, this won't work, it should :wink: 

Maybe (not tested):
```suggestion
                return a.replace(/[&?]media=/, "") + c;
```

Reply:
> If media is the first URL parameter, this won't work

Indeed, it's fixed.

> Code suggestion

I think this will remove the `?`. As I'm not a fan of regEx, I preferred to use `URLSearchParams` to remove the param.

So it's ready for your review again @qsm-odoo .